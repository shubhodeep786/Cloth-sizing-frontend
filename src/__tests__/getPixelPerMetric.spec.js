const cv = require('opencv');
const { getContoursWithMutationToImage } = require('../createContoursStream');
const { getLeftMostContour, getPPM } = require('../getPixelPerMetric');

describe('getLeftMostContour', () => {
  test('test 1', (done) => {
    cv.readImage('./assets/jean_with_ref.jpg', (err, image) => {
      const contours = getContoursWithMutationToImage(image);
      const leftmost = getLeftMostContour(contours);
      expect(leftmost).toMatchSnapshot();
      done();
    })
  })
});

describe('getPPM', () => {
  test('test 1', () => {
    const refRect = {
      "height": 161,
      "width": 161,
      "x": 33,
      "y": 543,
    }
    expect(getPPM(refRect, .5, .5)).toMatchSnapshot();
  })

  test('test 2', () => {
    const refRect = {
      "height": 161,
      "width": 181,
      "x": 33,
      "y": 543,
    }
    expect(() => getPPM(refRect, .5, .5)).toThrowErrorMatchingSnapshot();
  })
});
