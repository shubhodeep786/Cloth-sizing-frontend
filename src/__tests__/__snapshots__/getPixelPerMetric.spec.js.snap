// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getLeftMostContour test 1 1`] = `
Object {
  "height": 161,
  "width": 161,
  "x": 33,
  "y": 543,
}
`;

exports[`getPPM test 1 1`] = `322`;

exports[`getPPM test 2 1`] = `"image is not a 90 deg view"`;
